blosc/frame.c

```diff
@@ -1243,6 +1310,169 @@ int frame_get_metalayers(blosc2_frame_s* frame, blosc2_schunk* schunk) {
   return ret;
 }
 
+static int frame_get_vlmetalayers_from_trailer(blosc2_frame* frame, blosc2_schunk* schunk, uint8_t* trailer,
+                                               int32_t trailer_len) {
+  int64_t trailer_pos = FRAME_TRAILER_VLMETALAYERS;
+
+  // Get the size for the index of metalayers
+  uint16_t idx_size;
+  trailer_pos += 2 + sizeof(idx_size);
+  if (trailer_len < trailer_pos) {
+    return BLOSC2_ERROR_READ_BUFFER;
+  }
+  big_store(&idx_size, trailer + FRAME_TRAILER_VLMETALAYERS + 2, sizeof(idx_size));
```

Comment:
These few lines might be the cause of the fuzzer issue. You should ensure that the `trailer + FRAME_TRAILER_VLMETALAYERS + 4` does not exceed the length of the frame.

Reply:
I have downloaded the artifact that CI Fuzz has uploaded. I understand that the `decompress_frame_fuzzer-testcase` is the file that produces the error and its contains a cframe plus it length at the end. Am I right? 

After some debugging using this file, I could not find anything wrong. I think the problem is in the file creation, but I'm not sure (the `msgpack2json` command line tool does not parse correctly the header of the `decompress_frame_fuzzer-testcase` file).

Anyway, this part of the code has been improved assuring that the length is checked. It would be nice if you can shed more light of the above error. Thanks!